### DISCUSSION TITLE
dialog and subprocess

### DISCUSSION THREAD

**Discussion by aud-jeremy** [zauberzeug/nicegui#2487] [Q&A] [answered]
### Description

Hello,

When i try to open a dialog box and after start a subprocess i never see the dialog.

For example with the code :
```py
from nicegui import ui
import subprocess

with ui.dialog() as dialog, ui.card():
    ui.label('Hello world!')
    ui.button('Close', on_click=dialog.close)

ui.button('Open a dialog', on_click=lambda e: buton())

def buton():
    dialog.open()
    res = subprocess.run([r"notepad.exe"], shell=True, capture_output=True, text=True)

ui.run()
```

Notepad is started but i didn't see the dialog.

I am on widowns 10 with python 3.11.2 and the version of nicegui : 1.4.13

Have a good day.

**Comment by falkoschindler** [comment, marked as answer]
Hi @aud-jeremy,

It looks like `subprocess.run` is blocking the main thread, which is why the UI isn't updated anymore.
Have a look at [this FAQ](https://github.com/zauberzeug/nicegui/wiki/FAQs#why-is-my-long-running-function-blocking-ui-updates) for more information and references.

**Reply by aud-jeremy** [reply]
Thanks for your answer.
I try to use some solution of the FAQ

**Reply by aud-jeremy** [reply]
With this example : https://github.com/zauberzeug/nicegui/blob/main/examples/ffmpeg_extract_images/main.py
I can find a solution of my problem.
Thanks a lot.
